lib/NOM/IO.hs

```diff
@@ -240,13 +252,47 @@ processTextStream config parser updater maintenance printerMay finalize initialS
       waitForInput = atomically $ check =<< readTVar refresh_display_var
   printerMay & maybe keepProcessing \(printer, output_handle) -> do
     linesVar <- newTVarIO 0
-    let writeToScreen :: IO ()
-        writeToScreen = writeStateToScreen (not config.silent) linesVar state_var output_builder_var refresh_display_var maintenance printer output_handle
+    let keepProcessingStdin :: IO ()
+        keepProcessingStdin = bracket setBuffering restoreBuffering $ const processStdinForeverLoop
```

Comment:
Sorry for not being more explicit about this, but we are already doing some terminal mode switching in Main.hs (grep for `hideCursor`, `showCursor`). We should strive for a unified solution.